src/generic_ui/scripts/ui.ts

```diff
@@ -341,6 +345,15 @@ module UI {
 
         this.updateSharingStatusBar_();
       });
+
+      core.onUpdate(uProxy.Update.FIRE_SIGNAL, this.fireSignal);
```

Comment:
Actually, for updates from the core, can we not use FIRE_SIGNAL?  Can we have something specific for each update, e.g. SHARE_ACCESS_FAILED, then the ui.ts can decide in which cases whether to call fireSignal or do something else?